File: csharp/hello_world/hello_world.cs
// ============================================================================
//  hello_world -- C# port of examples\Vb6\hello_world\hello_world.bas
//  Writes out.pdf with a single centered line of italic text plus a timestamp.
// ============================================================================
using System;
using System.IO;
using LumasPdfSdk;

class HelloWorld
{
    // Keep the delegate alive so the GC cannot collect it while the DLL holds it.
    static TErrorProc _err = ErrProc;

    static int ErrProc(IntPtr data, int errCode, string errMessage, int errType)
    {
        Console.WriteLine(errMessage);
        return -1;                       // break processing if an error occurs
    }

    static void Main()
    {
        IntPtr pdf = LumasPdf.pdfNewPDF();
        LumasPdf.pdfSetOnErrorProc(pdf, IntPtr.Zero, _err);
        if (!LumasPdf.pdfCreateNewPDFW(pdf, ""))     // output file opened later
        {
            LumasPdf.pdfDeletePDF(pdf);
            return;
        }
        LumasPdf.pdfSetDocInfoW(pdf, TDocumentInfo.diCreator, "Delphi Example project");
        LumasPdf.pdfSetDocInfoW(pdf, TDocumentInfo.diTitle, "My first PDF output");

        LumasPdf.pdfAppend(pdf);
        LumasPdf.pdfSetFontW(pdf, "Arial", LumasPdfConsts.fsItalic, 30.0, true, TCodepage.cp1252);
        LumasPdf.pdfWriteFTextW(pdf, LumasPdfConsts.taCenter,
            "My first PDF output..." + "\r" + "\r" + DateTime.Now.ToString());
        LumasPdf.pdfEndPage(pdf);

        string outFile = "";
        if (LumasPdf.pdfHaveOpenDoc(pdf))
        {
            LumasPdf.pdfSetOnErrorProc(pdf, IntPtr.Zero, null);
            outFile = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "out.pdf");
            if (!LumasPdf.pdfOpenOutputFileW(pdf, outFile))
            {
                LumasPdf.pdfDeletePDF(pdf);
                return;
            }
            LumasPdf.pdfSetOnErrorProc(pdf, IntPtr.Zero, _err);
        }
        if (LumasPdf.pdfCloseFile(pdf))
            Console.WriteLine("OK: " + outFile);

        LumasPdf.pdfDeletePDF(pdf);
    }
}

File: vbnet/hello_world/hello_world.vb
' hello_world -- VB.NET port of examples\Vb6\hello_world
Imports System
Imports System.IO
Imports LumasPdfSdk

Module HelloWorld
    ' Keep a reference so the delegate is not GC'd during native calls.
    Private errDel As TErrorProc

    Public Function ErrProc(ByVal Data As IntPtr, ByVal ErrCode As Integer, ByVal ErrMessage As String, ByVal ErrType As Integer) As Integer
        Console.WriteLine(ErrMessage)
        Return -1 ' break processing on error
    End Function

    Sub Main()
        Dim pdf As IntPtr = LumasPdf.pdfNewPDF()
        errDel = New TErrorProc(AddressOf ErrProc)
        LumasPdf.pdfSetOnErrorProc(pdf, IntPtr.Zero, errDel)

        If LumasPdf.pdfCreateNewPDFW(pdf, "") = 0 Then
            LumasPdf.pdfDeletePDF(pdf)
            Return
        End If
        LumasPdf.pdfSetDocInfoW(pdf, TDocumentInfo.diCreator, "Delphi Example project")
        LumasPdf.pdfSetDocInfoW(pdf, TDocumentInfo.diTitle, "My first PDF output")

        LumasPdf.pdfAppend(pdf)
        LumasPdf.pdfSetFontW(pdf, "Arial", LumasPdfConsts.fsItalic, 30.0, True, TCodepage.cp1252)
        LumasPdf.pdfWriteFTextW(pdf, LumasPdfConsts.taCenter, "My first PDF output..." & vbCr & vbCr & Now().ToString())
        LumasPdf.pdfEndPage(pdf)

        Dim outFile As String = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "out.pdf")
        If LumasPdf.pdfHaveOpenDoc(pdf) <> 0 Then
            LumasPdf.pdfSetOnErrorProc(pdf, IntPtr.Zero, Nothing)
            If LumasPdf.pdfOpenOutputFileW(pdf, outFile) = 0 Then
                LumasPdf.pdfDeletePDF(pdf)
                Return
            End If
            LumasPdf.pdfSetOnErrorProc(pdf, IntPtr.Zero, errDel)
        End If
        If LumasPdf.pdfCloseFile(pdf) <> 0 Then
            Console.WriteLine("OK: " & outFile)
        End If
        LumasPdf.pdfDeletePDF(pdf)
    End Sub
End Module
